lightning/src/ln/onchaintx.rs

```diff
@@ -0,0 +1,696 @@
+				&InputDescriptors::OfferedHTLC => {
+					1 + 1 + 73 + 1 + 32 + 1 + 133
+				},
+				// number_of_witness_elements + sig_length + revocation_sig + pubkey_length + revocationpubkey + witness_script_length + witness_script
+				&InputDescriptors::ReceivedHTLC => {
+					1 + 1 + 73 + 1 + 1 + 1 + 139
+				},
+				// number_of_witness_elements + sig_length + revocation_sig + true_length + op_true + witness_script_length + witness_script
+				&InputDescriptors::RevokedOutput => {
+					1 + 1 + 73 + 1 + 1 + 1 + 77
+				},
+			};
+		}
+		tx_weight
+	}
+
+	fn get_height_timer(current_height: u32, timelock_expiration: u32) -> u32 {
+		if timelock_expiration <= current_height || timelock_expiration - current_height <= 3 {
+			return current_height + 1
+		} else if timelock_expiration - current_height <= 15 {
+			return current_height + 3
+		}
+		current_height + 15
+	}
+
+	/// Lightning security model (i.e being able to redeem/timeout HTLC or penalize coutnerparty onchain) lays on the assumption of claim transactions getting confirmed before timelock expiration
+	/// (CSV or CLTV following cases). In case of high-fee spikes, claim tx may stuck in the mempool, so you need to bump its feerate quickly using Replace-By-Fee or Child-Pay-For-Parent.
+	fn generate_claim_tx(&self, height: u32, cached_claim_datas: &ClaimTxBumpMaterial, fee_estimator: &FeeEstimator) -> Option<(u32, u64, Transaction)> {
+		if cached_claim_datas.per_input_material.len() == 0 { return None } // But don't prune pending claiming request yet, we may have to resurrect HTLCs
+		let mut inputs = Vec::new();
+		for outp in cached_claim_datas.per_input_material.keys() {
+			log_trace!(self, "Outpoint {}:{}", outp.txid, outp.vout);
+			inputs.push(TxIn {
+				previous_output: *outp,
+				script_sig: Script::new(),
+				sequence: 0xfffffffd,
+				witness: Vec::new(),
+			});
+		}
+		let mut bumped_tx = Transaction {
+			version: 2,
+			lock_time: 0,
+			input: inputs,
+			output: vec![TxOut {
+				script_pubkey: self.destination_script.clone(),
+				value: 0
+			}],
+		};
+
+		macro_rules! RBF_bump {
+			($amount: expr, $old_feerate: expr, $fee_estimator: expr, $predicted_weight: expr) => {
+				{
+					let mut used_feerate;
+					// If old feerate inferior to actual one given back by Fee Estimator, use it to compute new fee...
+					let new_fee = if $old_feerate < $fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::HighPriority) {
+						let mut value = $amount;
+						if subtract_high_prio_fee!(self, $fee_estimator, value, $predicted_weight, used_feerate) {
+							// Overflow check is done in subtract_high_prio_fee
+							$amount - value
+						} else {
+							log_trace!(self, "Can't new-estimation bump new claiming tx, amount {} is too small", $amount);
+							return None;
+						}
+					// ...else just increase the previous feerate by 25% (because that's a nice number)
```

Comment:
(I know this is not changed in this PR)

Might want to consider escalating the fee much faster, in case the fee estimator is wrong.  Even switching to a doubling mode after some time.  The game theory is that we might lose the entire channel value, so even if we end up paying 10% in fees it may be worth it.

Also, if the fee estimator increases moderately for a long time, this branch will never be taken, and again if the fee estimator is wrong we can lose funds.

Reply:
I agree picking up a %25 bump was just magic and more aggressive heuristics could be implemented. So far IIRC we are the only implem bumping fees would like more feedback from other LN devs.

The game theory remark is a very good one.